### DISCUSSION TITLE
Aim for SPEC0 adherence over NEP29

### DISCUSSION THREAD

**Discussion by IAlibay** [MDAnalysis/mdanalysis#4466] [Development]
## Background

As outlined here: https://github.com/numpy/numpy/issues/24932#issue-1943808025 the [NEP29](https://numpy.org/neps/nep-0029-deprecation_policy.html) enhancement proposal has been superseded by [SPEC0](https://scientific-python.org/specs/spec-0000/).

There is very little difference between the two, except that SPEC0 has a shorter support window (3 years for Python, 2 years for core packages) than NEP29 (42 months for Python, and 2 years for NumPy). SPEC0 also defines support windows for a wider range of core packages (e.g. scipy, matplotlib, etc..).

## How would this affect MDAnalysis?

In practice this would have very little impact. The difference between NEP29 and SPEC0 is small enough that we would be unlikely to see anything given our release frequency.

The main thing would be that we would be basing decisions to drop support for Python & other upstream dependency versions on the SPEC0 schedule rather than the NEP29 one. It would also mean that the day I finally get to write [this blog post](https://github.com/MDAnalysis/MDAnalysis.github.io/issues/238) it would be about SPEC0, not NEP29.

## Does this mean we _have_ to strictly adhere to SPEC0?

The idea here is that SPEC0 defines the "minimum range of versions which should be supported", rather than "the only things that should be supported".

That means that we can support a wider range of dependency versions if we wish to. This would be up to the community to help define.

## What if we don't adopt SPEC0

We could stick to NEP29 or seek alternative schemes to define what range of core dependency versions we support.

**Comment by hmacdope** [comment]
If upstream is moving then we should move too is my 2p.

**Reply by orbeckst** [reply]
I suppose so... from a science perspective I very much dislike even shorter support windows.

I am not a fan of following the tech industries' "make everything obsolete as quickly as possible" approach. However, I've come to see this as the price for being part of the Python eco-system.

**Reply by orbeckst** [reply]
(And yes, we can choose to support for longer, but in practice this means more work for us.)

P.S.: Thank you for the executive summary @IAlibay !

**Comment by orbeckst** [comment]
I generally agree with @hmacdope that we should follow along with the broader community — maintaining compatibility beyond the community time frame is not a hill I want to die on. If anyone really needs the extended compatibility then they should ask for a support contract with MDAnalysis of some kind. Academic research groups might either have to factor software support into their grant applications or provide "in kind" support by encouraging students to contribute to open source software. 

So in summary, I am ok with formally adopting SPEC0.

@MDAnalysis/coredevs are there enough "yes" voices here that we can just adopt it or do we need to discuss at a business meeting? Or is everybody against it? It would be good to resolve this question and if we can do this asynchronously then that saves everybody time.

You can just leave a 👍 or 👎 on the original post.

**Comment by RMeli** [comment]
Happy to move from NEP29 to SPEC0

**Comment by orbeckst** [comment]
So we have 3 x 👍 in favor and no more votes rolling in so this means we will **aim for SPEC0 adherence over NEP29**.

**Reply by orbeckst** [reply]
I created #4536 so that we don't forget — anyone with insights please feel free to edit the issue.

**Comment by orbeckst** [comment]
I am closing the discussion, now that we have #4536
